builder/shell-for.nix

```diff
@@ -90,8 +90,15 @@ let
   # We need to remove any inputs which are selected components (see above).
   # `buildInputs`, `propagatedBuildInputs`, and `executableToolDepends` contain component
   # derivations, not packages, so we use `removeSelectedInputs`).
-  systemInputs = removeSelectedInputs (lib.concatMap
-    (c: c.buildInputs ++ c.propagatedBuildInputs) selectedComponents);
+  #
+  # Also, we take care to keep duplicates out of the list, otherwise we may see
+  # "Argument list too long" errors from bash when entering a shell.
+  uniqueInputs = inputList: builtins.listToAttrs (builtins.map (x: lib.nameValuePair (x.name) x) inputList);
+  unionComponentInputs = inputsSoFar: c:
+    inputsSoFar // uniqueInputs c.buildInputs // uniqueInputs c.propagatedBuildInputs;
+  systemInputs' = lib.lists.foldl unionComponentInputs {} selectedComponents;
+  systemInputs = removeSelectedInputs (builtins.attrValues systemInputs');
+
```

Comment:
Can drop this patch too if we bump our mainline revision https://github.com/input-output-hk/haskell.nix/pull/1613

Reply:
sweet, want to update your branch with that? should be able to merge upstream master, right now `ours` is identical to `master`